src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\MainControls;
+
+use \ILIAS\UI\Component\Image\Image;
+
+/**
+ * This is what a factory for main controls looks like.
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Metabar is a unique page section to accomodate elements that
+	 *     should permamently be in sight of the user.
+	 *     The Metabar shall, first of all, host Prompts, i.e. notifications
+	 *     from the System to the user, but may also accomodate components and
+	 *     links deemed important, like help or search. The content of the bar
+	 *     does not change when navigating the system, but may depend on a
+	 *     configuration.
+	 *
+	 *   composition: >
+	 *     The Metabar is rendered horizontally at the very top of the page. It
+	 *     is always visible and available (except in some specialized view modes
+	 *     like an exam mode) as a static screen element and is unaffected by
+	 *     scrolling.
+	 *
+	 *     The Metabar always features the logo on the left hand side, while
+	 *     further elements are placed on the right hand side. Currently, these
+	 *     are "Search", "Help", "Notifications", "Awareness" and "User"
+	 *
+	 *     Especially in mobile context, the total width of all entries may exceed
+	 *     the availble width of the screen. In this case, all entries are
+	 *     summarized under a "..."-Button. The logo remains.
+	 *
+	 *     Elements are rendered as Bulky Buttons. Prompts in the Metabar may be
+	 *     marked with counters for new/existing notifications.
+	 *
+	 *   effect: >
+	 *     Entries in the Metabar may open a Slate when clicked. They will be set
```

Comment:
This is where the confusion around metabar entries vs. bulky buttons vs. prompts and slates start for me.

The current interface allows to add bulky buttons or prompts to the metabar. However "Entries in the Metabar may open a Slate when clicked". This does no make sense for Prompts being added to the metabar. They do not open slates, they are slates. There is currently no example for using a Prompt in the metabar, maybe this will make this clear. But for me it remains unclear if the triggerer (bulky buttons) are part of the slates or not.

Reply:
As far as we understand the requirements for the metadata, there currently are two types of entries: one that just is a button that does something (aka "?", which adds an according slate to the tools-section of the mainbar) and buttons that open a slate. Since we decided that the label and icon/glyph should be attached to the slate, what a consumer of the interface in fact needs to add are buttons (for the first case) and slates, where in the second case the metabar internally uses the glyph/icon and the label from the slate to create an according button and wire it to a popover that contains the slate. Does this make sense?

Reply:
Yes, thanks. See also this comment: https://github.com/ILIAS-eLearning/ILIAS/pull/1335#discussion_r238260684